app/controllers/tasks_controller.rb

```diff
@@ -41,14 +41,15 @@ def destroy
 
   private
 
-  # Use callbacks to share common setup or constraints between actions.
   def set_task
     @task = Task.find(params[:id])
   end
 
-  # Only allow a trusted parameter "white list" through.
   def task_params
-    params.require(:data).require(:attributes).permit(task_valid_attributes)
+    params.require(:data)
+      .require(:attributes)
+      .transform_keys(&:underscore)
```

Comment:
So we should change the API definition, because the params don't go in underscore, shouldn't we?

This one, for example: https://github.com/singularities/circular-works/blob/master/api.raml#L45